src/hooks/useMutation.ts

```diff
@@ -0,0 +1,74 @@
+//libs
+import { useCallback, useState } from "react";
+
+//constants
+import { STATUS } from "../Constants/constants";
+
+type statusProps = {
+  status: string;
+  data: any;
+  error: Error | undefined;
+};
+
+export const useMutation = <TMutation>(url: string, config: any) => {
+  const [state, setState] = useState<statusProps>(() => {
+    return { status: STATUS.IDLE, data: undefined, error: undefined };
+  });
```

Comment:
We should only use initializer function if we are doing some computation to generate the initial state. In this case we can extract out the initial state as a constant.